runtime/src/qio/qio_popen.c

```diff
@@ -576,3 +578,78 @@ qioerr qio_proc_communicate(
   return err;
 }
 
+/* Send a signal to the specified pid.
+
+   The signal value `qio_sig` is not the actual signal value defined in
+   signal.h.  The qio_sig value is Chapel-specific and must be converted to
+   the system-specific signal value.
+ */
+qioerr qio_send_signal(int64_t pid, int qio_sig)
+{
+  qioerr err = 0;
+  int rc = 0;
+  int sig = 0;
+
+  // Convert qio_sig values (from Chapel signal enums) to system-specific
+  // signal values.  This isn't ideal, but we don't have extern params.
```

Comment:
The `extern const` approach seems reasonable (and a little less confusing overall).

I think I'm somewhat over-inclined to use `enum` types, but Chapel doesn't have `extern param` support, so I couldn't make the two values match exactly. I think the main benefit of using an `enum` over an `extern const` is that string conversion and printing converts on the name, not the value (and, for example, the value of SIGTERM is not immediately meaningful to me without connecting it to the name). Of course, even if Chapel did support `extern param` types, I don't know how it could handle an `enum` with names set to the `extern param` values for the non-OS-portable signals.